src/shapes/text.class.js

```diff
@@ -239,6 +240,15 @@
      */
     pathSide:               'left',
 
+    /**
+     * How text is aligned to the path. This property determines
+     * the perpendicular position of each character relative to the path.
+     * (one of "baseline", "center", "ascender", "descender")
+     * @type String
+     * @default
+     */
+    pathAlign:               'baseline',
```

Comment:
So this is actually the baseline, not the pathAlign, sort of pathBaseline.
We do not have much to calculate ascender and descender.
Maybe advanced text metrics.

Reply:
The terms I used are actually taken straight from from Illustrator/InDesign.

![image](https://user-images.githubusercontent.com/9057412/132767953-1d862b02-5de9-4400-938f-daf74bb32f12.png)

In typography, the baseline is defined as "the line upon which most letters "sit" and below which descenders extend" (taken from [Wikipedia](https://en.wikipedia.org/wiki/Baseline_(typography)))

![image](https://user-images.githubusercontent.com/9057412/132920951-209fc187-b032-4496-bf6e-1812fc24be29.png)


I'm not sure why canvas uses `textBaseline` as the general property name, but I personally feel like calling it the Baseline setting is a bit confusing. Thoughts?

Reply:
My only issue here is that each font family has those different. Especially the one that aren't latin fonts.
So baseline is the one we know and should work, then ewe can have descener and ascender using lineHeight, i m not sure we can get center working.

Reply:
i m open to test this as it is, but i wonder if changing the real baseline at render time ( for path case only ) would yeld better results and valid across different fonts families.

https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/textBaseline

changing this at render time, would be exactly 1 line and no magic numbers. But i m not sure of which results would give.
The reason why we could use it in this particular case compared to normal rendering is because in this cases the bounding box is broken most of the time.

Reply:
The textBaseline property does seem to give better results across fonts so if you're not bothered by the slight differences in browser rendering I'm fine with going that route instead.

Reply:
One problem though is that if we go with the textBaseline prop there's no way to use the lineheight for the offset distance since every character is aligned to its own vertical center. Having the baseline shift between characters of different styles seems odd to me, and doesn't match how InDesign/Illustrator treats text of different styles on a path.

![image](https://user-images.githubusercontent.com/9057412/134092924-8b8119fe-93fa-492a-b361-60ad1f29dc5a.png)

I'm conflicted on this. 😐